### ISSUE TITLE
Tesla  watch app

### ISSUE THREAD

**Issue by pwvw** [Kemmey/Kemmey-TeslaWatch-Public#151]
remote start is grayed out and does not work why?

**Comment by Kemmey** [comment]
Did you enable remote start in settings and input your password correctly?

**Comment by JHIndyY** [comment]
After upgrading to ios14 and latest Apple watch os. Watch app doesn't work??

**Comment by Kemmey** [comment]
@JHIndyY how doesn't work? Can you be more specific?

**Comment by JHIndyY** [comment]
Thanks for the reply.

Iphone X iOS 14. iWatch iOS ver 7.

It’s been working fine for the last month. Having trouble since I upgraded to iOS 14. Have not tried to delete and reinstall. Just continues to say “Loading”.


John

**Comment by Kemmey** [comment]
Could you verify which version you have installed - open settings and check at the top for version number. 

Where does it say loading - in the titlebar? Is the vehicle online in the tesla iOS app?

**Comment by JHIndyY** [comment]
The watch app is v 1.0.7 build 290.

Everything works in the iOS app on my phone.


It says “Loading (4d) in the title bar (left) with the time of day on the right side.

**Comment by Kemmey** [comment]
That's interesting - everything checks out. 

Could you try tapping the name of your vehicle and then choose it in the next menu - that should reset vehicle data. It might then just revert to say no data at all instead. But please leave it there a while. 

If still stuck, try force quitting the app by first holding the side button until the shutdown menu shows up, then hold in the crown until the app quits and you are back at the watch face. 

If still stuck, try signing out and back in.

**Comment by JHIndyY** [comment]
After choosing the vehicle name, it just says “Model Y loading”. I’ll leave it there for awhile as you suggested.

**Comment by JHIndyY** [comment]
Working now. Signed out and back in.

Thanks for the assistance.

**Comment by Kemmey** [comment]
Awesome! Really weird that it was stuck at loading, I'll have to look into that. There must be a bug hiding there somewhere. 

Did you have the app running fine for approx 45 days, by any chance?

**Comment by JHIndyY** [comment]
Almost exactly 45 days, it was just fine. What’s up with that?

**Comment by Kemmey** [comment]
Simple really - the token generated at login lasts for 45 days, after which the app should use the refresh token to get a fresh token and refresh token. 

I bet there's a bug in there, that possibly the token was refreshed too late or not at all. 

I'll give that part of the code a thorough read-through and see what I can find. 

Thanks for reporting and helping make the app better for everyone! :-D

**Comment by JHIndyY** [comment]
Ok. 1 last question for now. Homelink currently shows as Inactive in the app. How do I make it “Active”?

**Comment by joetomasone** [comment]
I wonder if the token refresh couldn't be completed due to the outage yesterday.  

Is there logic to retry it and/or report if it cannot be obtained?

**Comment by Kemmey** [comment]
@JHIndyY Homelink will only be active when the car reports being near your primary Homelink device. Generally, if you can activate it with the iOS app, it should be available. If not, please let me know.

**Comment by Kemmey** [comment]
@joetomasone That would certainly be the perfect storm, and plausible. But it might be a generic bug with token renewal. I would have expected to have more reports if it was widespread, but people might just sign out and back in and not think more of it...

The eternal loading might be related to token refresh - that there is a logic error there at it just gets stuck. 

The refresh token can be used only one single time - so if the call fails but the server picks it up, there's no recovery path.

**Comment by JHIndyY** [comment]
I believe I have to have the FSD “Summon” feature or Enhanced AP to utilize Homelink in the iOS app. ??

**Comment by joetomasone** [comment]
Yep.   That would be why logout/login works - new token and refresh token
are generated.    Maybe the app can detect a refresh failure and initiate
the logout on it's own after a certain period of time to avoid a hangup
that confuses users.

**Comment by Kemmey** [comment]
@JHIndyY Hey, sorry for the delayed reply. 

Yes - Homelink for some weird reason require FSD. Basically, if it works in the iOS app, it should work on the watch. If not, then not...

**Comment by Kemmey** [comment]
@joetomasone I'll definitely need to do something. It should already do logout if token is invalid. Must be the refresh scenario that's buggy

**Comment by JHIndyY** [comment]
Thanks for responding. Have a good day.
